lib/maths/CDecompositionComponent.cc

```diff
@@ -136,6 +137,21 @@ TDoubleDoublePr CDecompositionComponent::value(double offset, double n, double c
             return {m, m};
         }
 
+        // Sanity check that neither m nor sd are NaN values
+        // This is an extremely rare edge case as checks elsewhere prevent NaNs from
+        // being added to the models. As yet it is unknown how NaNs could be returned
+        // from the splines, only that they do occur.
+        // The returned interval will not contribute to the baseline.
+        if (maths::CMathsFuncs::isNan(m)) {
+            LOG_WARN(<< "Calculated mean value is not a number.");
+            return {0.0, 0.0};
+        }
+
+        if (maths::CMathsFuncs::isNan(sd)) {
+            LOG_WARN(<< "Calculated standard deviation value is not a number.");
+            return {0.0, 0.0};
```

Comment:
`{m, m}` might be a better fallback here than `{0.0, 0.0}` - that's what we're returning elsewhere for edge cases.

Reply:
Yes, that was my first thought here too @droberts195, but it is possible that both `m` and `sd` are NaN values (which in itself is interesting as it indicates that the NaN values stem from both the linear and the cubic spline types). Returning `{0.0, 0.0}` here effectively means that we return `value` from `CTimeSeriesDecomposition::detrend` - the same as we would if not initialised.